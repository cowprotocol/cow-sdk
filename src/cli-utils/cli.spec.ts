import { exec } from 'child_process'

const pk = '0x54574aba486dbad0c4d3e7fced10cfa35331c51a4968a5e381723e5f43b819a2'

describe('Cow SDK CLI', () => {
  it('Sign order', (done) => {
    exec(
      `
    yarn run run-cli --silent -- \\
      --private-key="${pk}" \\
      --chainId=5 \\
      --operation=signOrder \\
      --kind=sell \\
      --partiallyFillable=false \\
      --expiresIn=600 \\
      --sellToken="0xB4FBF271143F4FBf7B91A5ded31805e42b2208d6" \\
      --buyToken="0x91056D4A53E1faa1A84306D4deAEc71085394bC8" \\
      --sellAmount="90000000000000000" \\
      --buyAmount="539761635000000000000" \\
      --feeAmount="0" \\
      --receiver=""
  `,
      (err, stdOut) => {
        const order = JSON.parse(stdOut.trim())

        delete order.validTo
        delete order.signature
        expect(order).toMatchSnapshot()
        done()
      }
    )
  }, 10_000)

  it('Send order', (done) => {
    exec(
      `
    yarn run run-cli --silent -- \\
      --private-key="${pk}" \\
      --chainId=5 \\
      --operation=sendOrder \\
      --kind=sell \\
      --partiallyFillable=false \\
      --expiresIn=600 \\
      --sellToken="0xB4FBF271143F4FBf7B91A5ded31805e42b2208d6" \\
      --buyToken="0x91056D4A53E1faa1A84306D4deAEc71085394bC8" \\
      --sellAmount="90000000000000000" \\
      --buyAmount="539761635000000000000" \\
      --feeAmount="0" \\
      --receiver=""
  `,
      (err, stdOut) => {
        expect(stdOut.trim()).toBe('The account needs to approve the selling token in order to trade')
        done()
      }
    )
  }, 10_000)

  it('Cancel order', (done) => {
    exec(
      `
    yarn run run-cli --silent -- \\
      --private-key="${pk}" \\
      --chainId=5 \\
      --operation=cancelOrder \\
      --orderUid="0xe0759f514e5e0a5d3e0ae7b492de87d72d765a94c4a6d16c5301a02c91cfaa44fb3c7eb936caa12b5a884d612393969a557d430763be64a2"
  `,
      (err, stdOut) => {
        expect(stdOut.trim()).toBe('The order you are trying to cancel does not exist')
        done()
      }
    )
  }, 10_000)

  it('Get order', (done) => {
    exec(
      `
    yarn run run-cli --silent -- \\
      --chainId=5 \\
      --operation=getOrder \\
      --orderUid="0xf499f5d64bbc49b744f03418ac70d4f2e02d71e235e64eaf51488741d2c2cffcd02de8da0b71e1b59489794f423fabba2adc4d93ffffffff"
  `,
      (err, stdOut) => {
        expect(stdOut.trim()).toMatchSnapshot()
        done()
      }
    )
  }, 10_000)

  it('Get orders', (done) => {
    exec(
      `
    yarn run run-cli --silent -- \\
      --chainId=5 \\
      --operation=getOrders \\
      --limit=2 \\
      --offset=0 \\
      --owner="0xfb3c7eb936caa12b5a884d612393969a557d4307"
  `,
      (err, stdOut) => {
        expect(stdOut.trim()).toMatchSnapshot()
        done()
      }
    )
  }, 10_000)
})
