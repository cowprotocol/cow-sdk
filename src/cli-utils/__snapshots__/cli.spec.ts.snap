// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cow SDK CLI Get order 1`] = `"{\\"creationDate\\":\\"2023-01-05T10:09:36Z\\",\\"owner\\":\\"0xfb3c7eb936caa12b5a884d612393969a557d4307\\",\\"uid\\":\\"0xf499f5d64bbc49b744f03418ac70d4f2e02d71e235e64eaf51488741d2c2cffcd02de8da0b71e1b59489794f423fabba2adc4d93ffffffff\\",\\"availableBalance\\":null,\\"executedBuyAmount\\":\\"47202130440337517731577833653\\",\\"executedSellAmount\\":\\"10100000000000000\\",\\"executedSellAmountBeforeFees\\":\\"7746436067977701\\",\\"executedFeeAmount\\":\\"2353563932022299\\",\\"invalidated\\":false,\\"status\\":\\"fulfilled\\",\\"class\\":\\"market\\",\\"settlementContract\\":\\"0x9008d19f58aabd9ed0d60971565aa8510560ab41\\",\\"fullFeeAmount\\":\\"2329694857061196\\",\\"ethflowData\\":{\\"userValidTo\\":1672915156,\\"refundTxHash\\":null,\\"isRefunded\\":false},\\"onchainOrderData\\":{\\"sender\\":\\"0xfb3c7eb936caa12b5a884d612393969a557d4307\\",\\"placementError\\":null},\\"onchainUser\\":\\"0xfb3c7eb936caa12b5a884d612393969a557d4307\\",\\"isLiquidityOrder\\":false,\\"sellToken\\":\\"0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE\\",\\"buyToken\\":\\"0xdc31ee1784292379fbb2964b3b9c4124d8f89c60\\",\\"receiver\\":\\"0xfb3c7eb936caa12b5a884d612393969a557d4307\\",\\"sellAmount\\":\\"7746436067977701\\",\\"buyAmount\\":\\"46455599245572845358066319128\\",\\"validTo\\":1672915156,\\"appData\\":\\"0xe7735f974c61ba027332fb9536ed844c153d77ed593d026548bbdc3cf3a904c4\\",\\"feeAmount\\":\\"2353563932022299\\",\\"kind\\":\\"sell\\",\\"partiallyFillable\\":false,\\"sellTokenBalance\\":\\"erc20\\",\\"buyTokenBalance\\":\\"erc20\\",\\"signingScheme\\":\\"eip1271\\",\\"signature\\":\\"0xd02de8da0b71e1b59489794f423fabba2adc4d93\\",\\"interactions\\":{\\"pre\\":[{\\"target\\":\\"0xd02de8da0b71e1b59489794f423fabba2adc4d93\\",\\"value\\":\\"0x0\\",\\"callData\\":[76,132,193,200]}]},\\"totalFee\\":\\"2353563932022299\\"}"`;

exports[`Cow SDK CLI Get orders 1`] = `"[{\\"creationDate\\":\\"2023-01-11T07:16:27.250763Z\\",\\"owner\\":\\"0xfb3c7eb936caa12b5a884d612393969a557d4307\\",\\"uid\\":\\"0xe0759f514e5e0a5d3e0ae7b492de87d72d765a94c4a6d16c5301a02c91cfaa44fb3c7eb936caa12b5a884d612393969a557d430763be64a2\\",\\"availableBalance\\":null,\\"executedBuyAmount\\":\\"0\\",\\"executedSellAmount\\":\\"0\\",\\"executedSellAmountBeforeFees\\":\\"0\\",\\"executedFeeAmount\\":\\"0\\",\\"invalidated\\":true,\\"status\\":\\"cancelled\\",\\"class\\":\\"limit\\",\\"surplusFee\\":\\"28446625236906\\",\\"surplusFeeTimestamp\\":\\"2023-01-11T07:16:29.105885Z\\",\\"executedSurplusFee\\":null,\\"settlementContract\\":\\"0x9008d19f58aabd9ed0d60971565aa8510560ab41\\",\\"fullFeeAmount\\":\\"31607361374340\\",\\"isLiquidityOrder\\":false,\\"sellToken\\":\\"0xb4fbf271143f4fbf7b91a5ded31805e42b2208d6\\",\\"buyToken\\":\\"0x91056d4a53e1faa1a84306d4deaec71085394bc8\\",\\"receiver\\":\\"0xfb3c7eb936caa12b5a884d612393969a557d4307\\",\\"sellAmount\\":\\"90000000000000000\\",\\"buyAmount\\":\\"539761635000000000000\\",\\"validTo\\":1673421986,\\"appData\\":\\"0x0000000000000000000000000000000000000000000000000000000000000000\\",\\"feeAmount\\":\\"0\\",\\"kind\\":\\"sell\\",\\"partiallyFillable\\":false,\\"sellTokenBalance\\":\\"erc20\\",\\"buyTokenBalance\\":\\"erc20\\",\\"signingScheme\\":\\"eip712\\",\\"signature\\":\\"0x94e341d6ba4bd19a03f44511be7ed34d90a57f08a1e0f0f1d389ec390360ff0b163ff856a1909285a9d14d7331066c47be20baf1d4c791c31f28ac4080186e5b1c\\",\\"interactions\\":{\\"pre\\":[]},\\"totalFee\\":\\"0\\"},{\\"creationDate\\":\\"2023-01-11T07:01:08.189981Z\\",\\"owner\\":\\"0xfb3c7eb936caa12b5a884d612393969a557d4307\\",\\"uid\\":\\"0x38c6a80ebe1baa6a7e18f06b251e8687c3b390cff83da4dbaebfd9e12b09fa10fb3c7eb936caa12b5a884d612393969a557d430763be610b\\",\\"availableBalance\\":null,\\"executedBuyAmount\\":\\"0\\",\\"executedSellAmount\\":\\"0\\",\\"executedSellAmountBeforeFees\\":\\"0\\",\\"executedFeeAmount\\":\\"0\\",\\"invalidated\\":false,\\"status\\":\\"expired\\",\\"class\\":\\"limit\\",\\"surplusFee\\":\\"20871465832731\\",\\"surplusFeeTimestamp\\":\\"2023-01-11T07:10:18.763566Z\\",\\"executedSurplusFee\\":null,\\"settlementContract\\":\\"0x9008d19f58aabd9ed0d60971565aa8510560ab41\\",\\"fullFeeAmount\\":\\"23190517591923\\",\\"isLiquidityOrder\\":false,\\"sellToken\\":\\"0xb4fbf271143f4fbf7b91a5ded31805e42b2208d6\\",\\"buyToken\\":\\"0x91056d4a53e1faa1a84306d4deaec71085394bc8\\",\\"receiver\\":\\"0xfb3c7eb936caa12b5a884d612393969a557d4307\\",\\"sellAmount\\":\\"90000000000000000\\",\\"buyAmount\\":\\"539761635000000000000\\",\\"validTo\\":1673421067,\\"appData\\":\\"0x0000000000000000000000000000000000000000000000000000000000000000\\",\\"feeAmount\\":\\"0\\",\\"kind\\":\\"sell\\",\\"partiallyFillable\\":false,\\"sellTokenBalance\\":\\"erc20\\",\\"buyTokenBalance\\":\\"erc20\\",\\"signingScheme\\":\\"eip712\\",\\"signature\\":\\"0x5c7890cec8217ea14091df6b8b7af0ed62272d2b228a6518fb55fbed9e4591e95b2ed35e560d26830919676a390742e6ff7f0a07b458e1d5704763b212ff5d891c\\",\\"interactions\\":{\\"pre\\":[]},\\"totalFee\\":\\"0\\"}]"`;

exports[`Cow SDK CLI Sign order 1`] = `"{\\"sellToken\\":\\"0xB4FBF271143F4FBf7B91A5ded31805e42b2208d6\\",\\"buyToken\\":\\"0x91056D4A53E1faa1A84306D4deAEc71085394bC8\\",\\"receiver\\":\\"0x96a3f0dc1BBBdFCB848782330C58133E0570B2DF\\",\\"sellAmount\\":\\"90000000000000000\\",\\"buyAmount\\":\\"539761635000000000000\\",\\"validTo\\":1673425310,\\"feeAmount\\":\\"0\\",\\"kind\\":\\"sell\\",\\"partiallyFillable\\":false,\\"signature\\":\\"0x7ace4a07e7ad7646bbc7c5b0faf2d8dd5606f05b2ea65fa8846495b85e12c979747688707d329494cf107541b59dabc509a3dcdf84b222dd9630cbd34bb5ba7e1b\\",\\"signingScheme\\":0}"`;
